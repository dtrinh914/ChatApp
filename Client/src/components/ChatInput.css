.ChatInput{
    height: 10vh;
    display: flex;
}
.ChatInput textarea{
    flex-grow: 4;
    resize: none;
}
.ChatInput button{
    flex-grow: 1;
}