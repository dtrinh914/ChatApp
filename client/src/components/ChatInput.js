import React, {useEffect} from 'react';
import useInput from '../hooks/useInput';
import Button from '@material-ui/core/Button';
import {makeStyles} from '@material-ui/styles'

const useStyles = makeStyles({
    root:{
        display: 'flex',
        height: '40px',
    },
    input:{
        fontSize: '1rem',
        flexGrow: '1'
    },
    button:{
        borderRadius:0
    }
})

function ChatInput({onConfirm, selected}){
    const classes = useStyles();
    const [message, setMessage, clearMessage] = useInput();

    useEffect(()=>{
        clearMessage();
        //eslint-disable-next-line
    },[selected]);
    function handleClick(){
        onConfirm(message);
        clearMessage();
    }
    
    function handleEnter(e){
        if(e.key === 'Enter'){
            e.preventDefault();
            handleClick();
        }
    }

    return(
        <div className={classes.root}>
            <input className={classes.input} type="text" value={message} onChange={setMessage} onKeyPress={handleEnter} />
            <Button className={classes.button} color="primary" variant="contained" onClick={handleClick}>Send</Button>
        </div>
    );
}

export default ChatInput;