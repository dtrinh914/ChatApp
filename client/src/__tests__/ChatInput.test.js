import React from 'react';
import ChatInput from '../components/ChatInput';
import { render, fireEvent, getByTestId, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup);

test('should match snapshot', () => {
    const {container} = render(<ChatInput />); 
    expect(container).toMatchSnapshot();
});

test('should display hello in input', () => {
    const {container} = render(<ChatInput />);
    const messageInput = getByTestId(container, 'message-input');
    fireEvent.change(messageInput ,{target: {value:'hello'}});
    expect(messageInput).toHaveValue('hello');
});

test('should send message and clear input', () => {
    const {container} = render(<ChatInput onConfirm={()=>{}} />);
    const messageInput = getByTestId(container, 'message-input');
    const sendButton = getByTestId(container, 'send-button');
    fireEvent.change(messageInput ,{target: {value:'hello'}});
    fireEvent.click(sendButton);

    expect(messageInput).toHaveValue('');
});