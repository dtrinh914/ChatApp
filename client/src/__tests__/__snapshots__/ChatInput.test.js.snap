// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <input
      class="makeStyles-input-2"
      data-testid="message-input"
      type="text"
      value=""
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-containedPrimary"
      data-testid="send-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Send
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
